src/runtime/str.cpp

```diff
@@ -341,7 +341,11 @@ extern "C" Box* strAdd(BoxedString* lhs, Box* _rhs) {
         return rtn;
     }
 
-    if (!PyString_Check(_rhs)) {
+    if (PyByteArray_Check(_rhs)) {
```

Comment:
super minor, but could we move this block inside the `if (!PyString_Check)` block?  I doubt this happens very often so it'd be nice to try avoid doing the check.